src/main/java/com/scriptbasic/syntax/commands/AbstractCommandAnalyzer.java

```diff
@@ -93,17 +93,16 @@ protected boolean isKeyWord(final String keyword) throws AnalysisException {
     }
 
     /**
-     * Checks that there are no extra characters on a program line when the line
-     * analyzer thinks that it has finished analyzing the line. If there are
+     * Checks that there are no extra characters on when the line
+     * analyzer thinks that it has finished analyzing the statement. If there are
      * some extra characters on the line then throws syntax error exception.
-     * Otherwise it simply steps the lexical analyzer iterator over the EOL
-     * symbol.
+     * Otherwise it simply steps the lexical analyzer iterator over the symbol.
      *
      * @throws AnalysisException when there are extra character on the actual line
      */
-    protected void consumeEndOfLine() throws AnalysisException {
+    protected void consumeEndOfStatement() throws AnalysisException {
         final var le = ctx.lexicalAnalyzer.get();
-        if (le != null && !le.isLineTerminator()) {
+        if (le != null && !(le.isLineTerminator()||le.isStatementSeparator())) {
```

Comment:
please leave spaces around the `||` operator to have consistent formatting with the rest of the code. If you ask IntelliJ to format the code using the default formatting rules that will be okay.